packages/jest-worker/src/index.ts

```diff
@@ -132,13 +137,12 @@ export default class JestWorker {
     return this._workerPool.getStdout();
   }
 
-  end(): void {
+  end(): Promise<PoolExitResult> {
     if (this._ending) {
       throw new Error('Farm is ended, no more calls can be done to it');
```

Comment:
should make this `end` `async` so this `throw` rejects a promise rather then throwing. Sucks to have to deal with both sync and async errors

Reply:
Done. Had to change a `index.test.js` test case to check for reject instead of throw but I guess it's an invalid call so not really breaking.